File: leetcode/28.find-the-index-of-the-first-occur-in-a-string/ts/strstr.test.ts
import { test, expect } from "vitest"
import strStr from "./strstr.js"

interface testcase {
    haystack: string
    needle: string
    want: number
}

test("strstr", () => {
    const tcs: testcase[] = [
        { haystack: "sadbutsad", needle: "sad", want: 0 },
        { haystack: "leetcode", needle: "leeto", want: -1 },
    ]

    tcs.forEach(tc => expect(strStr(tc.haystack, tc.needle) === tc.want))
})

File: leetcode/28.find-the-index-of-the-first-occur-in-a-string/go/strstr_test.go
package strstr

import "testing"

type testcase struct {
	haystack string
	needle   string
	want     int
}

func TestStrStr(t *testing.T) {
	tcs := []testcase{
		{haystack: "sadbutsad", needle: "sad", want: 0},
		{haystack: "leetcode", needle: "leeto", want: -1},
	}

	for _, tc := range tcs {
		t.Run("strstr", func(t *testing.T) {
			got := strStr(tc.haystack, tc.needle)
			if got != tc.want {
				t.Fatalf("strstr, want %d, got %d", tc.want, got)
			}
		})
	}
}
